src/types/stardust/block/output/native_token.rs

```diff
@@ -95,17 +95,31 @@ impl TryFrom<TokenScheme> for bee::TokenScheme {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct NativeToken {
     pub token_id: NativeTokenId,
     pub amount: NativeTokenAmount,
+    pub float_amount: f64,
+}
+
+impl PartialEq for NativeToken {
+    fn eq(&self, other: &Self) -> bool {
+        self.token_id == other.token_id && self.amount == other.amount
+    }
 }
+impl Eq for NativeToken {}
 
 impl From<&bee::NativeToken> for NativeToken {
     fn from(value: &bee::NativeToken) -> Self {
+        let amount = NativeTokenAmount::from(value.amount());
+        let (high, low) = (
+            u128::from_be_bytes(amount.0[..16].try_into().unwrap()),
+            u128::from_be_bytes(amount.0[16..].try_into().unwrap()),
+        );
         Self {
             token_id: NativeTokenId(**value.token_id()),
-            amount: value.amount().into(),
+            amount,
+            float_amount: ((high as f64) * 2f64.powi(16)) + low as f64,
```

Comment:
I don't think we should change the data model for this right now. I'm keeping a backlog of breaking database changes in the [Shimmer RC tracking issue](#486). Maybe you could add an item to the list there?

Reply:
If we don't change the data model, we cannot provide these values at all.